---
title: "R_rrbs_analysis_latest"
author: "Joselynn"
date: "7/22/2021"
output: html_document
---
# Some genes have duplicate rows -- for example, ENSSSCG00000017809 is appearing more than once. Where are they coming from? Is it from the Entrez gene IDs? 

# Transfac is a DB with transcription factor binding sites -- any others for porcine?

# Issue #2 is with the comparisons -- some of them are pretty similar, maybe de-duplicating will help.

# print the pivot_longer tables to CSV and upload them to the google drive just incase folks want to make their own heatmaps.

# in some cases, the m-value is 0. What does this mean? in some cases, if the methylated and unmethylated counts are 0, the m-value is still 0 but we should distinguish between these low coverage samples where there's no data (ND).

# Add some bubble or volcano plots to show the absolute expression vlue vs foldchange/mvalue




```{r, include = FALSE}
library("edgeR")
library("bsseq")
library("dplyr")
library("tidyr")
library("openxlsx")
library("biomaRt")
library("dplyr")
library("GenomicRanges")
library("stringr")
```

```{r}
targets <- read.table('/home/rstudio/dat/targets.txt', header = T, sep = '\t')
files <- targets$file             
yall <- readBismark2DGE(files, sample.names=targets$sample)

save.image('/home/rstudio/dat/session1.Rdata')
```
```{r}
load('/home/rstudio/dat/session1.Rdata')


yall <- yall[yall$genes[,'Chr'] == 1 | yall$genes[,'Chr'] == 2| yall$genes[,'Chr'] == 3| yall$genes[,'Chr'] == 4| yall$genes[,'Chr'] == 5| yall$genes[,'Chr'] == 6| yall$genes[,'Chr'] == 7| yall$genes[,'Chr'] == 8| yall$genes[,'Chr'] == 9| yall$genes[,'Chr'] == 10| yall$genes[,'Chr'] == 11| yall$genes[,'Chr'] == 12| yall$genes[,'Chr'] == 13| yall$genes[,'Chr'] == 14| yall$genes[,'Chr'] == 15| yall$genes[,'Chr'] == 16| yall$genes[,'Chr'] == 17| yall$genes[,'Chr'] == 18| yall$genes[,'Chr'] == 'X'| yall$genes[,'Chr'] == 'Y',]

sort_order <- c(1:18, 'X', 'Y')

yall$genes$Chr <- factor(yall$genes$Chr, levels=sort_order)

o <- order(yall$genes$Chr, yall$genes$Locus)

yall <- yall[o,]        

table(yall$genes$Chr)

#      1       2       3       4       5       6       7       8       9      10      11      12      13      14      15      16      17      18       X 
#2351408 1881377 2159242 1592211 1349283 2801759 1660304 1306871 1540861 1060010 1051247 1440491 1592620 1736525 1196939  788249 1000426  918874  872986 
#      Y 
#  43712 

save.image('/home/rstudio/dat/session2.Rdata')
```
```{r}
load('/home/rstudio/dat/session2.Rdata')

# make factor levels of Me and Un repeated along the length of the columns 
Methylation <- gl(2,1,ncol(yall), labels=c("Me","Un"))
Me <- yall$counts[, Methylation=="Me"]
Un <- yall$counts[, Methylation=="Un"]

# combine methylated and unmethylated reads to get total coverage
Coverage <- Me + Un

# Get a list of CpGs with at least 5 counts (methylated and unmethylated) in 3 samples 
HasCoverage <- rowSums(Coverage >= 5) == 3

# List of CpGs that are not 0 both
# this filters out CpG that are never methylated or always methylated as this provides no info about differential methylation 
HasBoth <- rowSums(Me) > 0 & rowSums(Un) > 0 

# Check sample size after filtering requirements are met 
table(HasCoverage, HasBoth)

save.image('/home/rstudio/dat/session3.Rdata')
```
```{r}
load('/home/rstudio/dat/session3.Rdata')

# apply those coverage filters
y <- yall[HasCoverage & HasBoth,, keep.lib.sizes=FALSE] 

# assign library sizes to each sample to be a sum of methylated and unmethylated counts
TotalLibSize <- y$samples$lib.size[Methylation=="Me"] + y$samples$lib.size[Methylation=="Un"]
y$samples$lib.size <- rep(TotalLibSize, each=2)
y$samples

save.image('/home/rstudio/dat/session4.Rdata')
table(y$genes$Chr)

#   1     2     3     4     5     6     7     8     9    10    11    12    13 
# 5275  6450  8154  4564  4175 10654  4802  3292  4270  3174  3831  6439  3177 
#   14    15    16    17    18     X     Y 
# 4443  2939  1952  3530  3203   524   151 
```
```{r}
load('/home/rstudio/dat/session4.Rdata')

#get all the genes used in  the DML analysis..
y_loc <- as.data.frame(y$genes)
y_gr <- makeGRangesFromDataFrame(y_loc,ignore.strand = T, start.field = 'Locus', end.field = 'Locus', seqnames.field = 'Chr')

#get granges for all tss in genome..

ss_mart <- useMart("ensembl", dataset = "sscrofa_gene_ensembl")

#attributes <- listAttributes(ss_mart)
#attributes[grep("symbol", attributes$description, ignore.case = TRUE), ]
#attributes[grep("transcription", attributes$description, ignore.case = TRUE), ]
#attributes[grep("vgnc", attributes$description, ignore.case = TRUE), ]

ss_tss <- getBM(attributes = c("transcription_start_site", 
                               "chromosome_name", 
                               "start_position",
                               "end_position",
                               "ensembl_gene_id", 
                               "strand", 
                               "uniprot_gn_symbol"), 
                mart = ss_mart)

ss_tss_2 <- getBM(attributes = c("transcription_start_site", 
                                 "chromosome_name", 
                                 "start_position",
                                 "end_position",
                                 "hgnc_symbol",
                                 "entrezgene_id",
                                 "description"), 
                  mart = ss_mart)

ss_tss <- as.data.frame(ss_tss)
ss_tss_2 <- as.data.frame(ss_tss_2)
ss_tss_all <- inner_join(ss_tss, ss_tss_2, by = c("transcription_start_site", "chromosome_name", "start_position", "end_position"))
ss_tss <- ss_tss_all

#find and replace the 'strand' column so that -1 is - and 1 is +
library(stringr)
ss_tss$strand <- str_replace(string = ss_tss$strand, pattern = '-1', replacement = '-')
ss_tss$strand <- str_replace(string = ss_tss$strand, pattern = '1', replacement = '+')

#filter so we only have chromosomes
ss_tss_filt <- dplyr::filter(ss_tss, chromosome_name=='1' |  chromosome_name=='2' |chromosome_name=='3' |chromosome_name=='4' |chromosome_name=='5' |chromosome_name=='6' |chromosome_name=='7' |chromosome_name=='8' |chromosome_name=='9' |chromosome_name=='10' |chromosome_name=='11' |chromosome_name=='12' |chromosome_name=='13' |chromosome_name=='14' |chromosome_name=='15' |chromosome_name=='16' |chromosome_name=='17' |chromosome_name=='18' |chromosome_name=='X' |chromosome_name=='Y')
ss_tss <- ss_tss_filt

# turn to granges
ss_tss_gr <- makeGRangesFromDataFrame(ss_tss, start.field = 'start_position', end.field = 'end_position', strand.field = 'strand')

# add metadata
ss_tss_gr$transcription_start_site <- ss_tss$transcription_start_site
ss_tss_gr$ensembl_gene_id <- ss_tss$ensembl_gene_id
ss_tss_gr$uniprot_gn_symbol <- ss_tss$symbol
ss_tss_gr$embl <- ss_tss$embl
ss_tss_gr$hgnc_symbol <- ss_tss$hgnc_symbol
ss_tss_gr$uniprot_gn_symbol <- ss_tss$uniprot_gn_symbol
ss_tss_gr$entrezgene_id <- ss_tss$entrezgene_id

save.image('/home/rstudio/dat/session5.Rdata')
```
```{r}
load('/home/rstudio/dat/session5.Rdata')

#distancetonearest to figure out how far away from a TSS each locus is (use this later to group into promoters)
dtn <- as.data.frame(distanceToNearest(x = y_gr, subject = ss_tss_gr))

#slice to get the subject/tss hits:
tss_slice <- as.data.frame(ss_tss_gr) %>% slice(dtn$subjectHits)

#add the 'distance' column to tss_slice:
tss_slice$distance <- dtn$distance

#add all the tss info to the rrbs data:
y$genes$nearest_gene_start <- tss_slice$start
y$genes$nearest_gene_end <- tss_slice$end
y$genes$nearest_gene_strand <- tss_slice$strand
y$genes$nearest_transcription_start_site <- tss_slice$transcription_start_site
y$genes$distance_to_nearest_tss <- tss_slice$distance
y$genes$ensembl_gene_id <- tss_slice$ensembl_gene_id
y$genes$hgnc_symbol <- tss_slice$hgnc_symbol
y$genes$entrezgene_id <- tss_slice$entrezgene_id

#get loci that are in promoters (e.g., less than 2kb away from TSS)
InPromoter <- y$genes$distance_to_nearest_tss <= 2000

#subset to only get loci in promoters
yIP <- y[InPromoter,,keep.lib.sizes=FALSE]

#get total counts for each promoter region
ypr <- rowsum(yIP, yIP$genes$ensembl_gene_id, reorder=FALSE)

#re-set library sizes based to be the average of the total read counts for the methylated and unmethylated libraries.

Me <- ypr$counts[, Methylation=="Me"]
Un <- ypr$counts[, Methylation=="Un"]

TotalLibSizepr <- 0.5*ypr$samples$lib.size[Methylation=="Me"] + 0.5*ypr$samples$lib.size[Methylation=="Un"]
ypr$samples$lib.size <- rep(TotalLibSizepr, each=2)
ypr$samples

#Me <- ypr$counts[, Methylation=="Me"]
#Un <- ypr$counts[, Methylation=="Un"]

M01 <- data.frame(log((Me + .01) / (Un + .01)))
M01$ensembl_gene_id <- rownames(M01)

M01 <- M01 %>% rename_all( ~ str_replace(., ".Me", "_M-value"))
M01 <- M01 %>% rename_all( ~ str_replace(., "RH.", "RH-"))

M01_log2 <- data.frame(log2((Me + .01) / (Un + .01)))
M01_log2$ensembl_gene_id <- rownames(M01_log2)

M01_log2 <- M01_log2 %>% rename_all( ~ str_replace(., ".Me", "_M-value"))
M01_log2 <- M01_log2 %>% rename_all( ~ str_replace(., "RH.", "RH-"))


save.image('/home/rstudio/dat/session6.Rdata')
```
```{r}
#In microarray methylation studies, a common measure of methylation level is the M-value, which is defined as M = log 2 {(Me + α) /(Un + α)} where Me and Un are the methylated and unmethylated intensities and α is some suitable offset to avoid taking logarithms of zero 39. The M-value can be interpreted as the base2 logit transformation of the proportion of methylated signal at each locus. (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5747346/)

#load('/home/rstudio/dat/session6.Rdata')

# make some output tables for later..
#library(dplyr)

counts_and_genes <- inner_join(tibble::rownames_to_column(as.data.frame(ypr$counts), var = 'loci'), tibble::rownames_to_column(as.data.frame(ypr$genes), var = 'loci'))
counts_and_genes$nearest_gene_start <- NULL
counts_and_genes$nearest_gene_end <- NULL
counts_and_genes$nearest_gene_strand <- NULL
counts_and_genes$loci <- NULL
counts_and_genes$Chr <- NULL

counts_and_genes <- inner_join(counts_and_genes, M01)

counts_and_genes <- counts_and_genes %>% relocate(ensembl_gene_id, .before = 'RH-1-Me')
counts_and_genes <- counts_and_genes %>% relocate(hgnc_symbol, .before = 'RH-1-Me')

counts_and_genes <- counts_and_genes %>% relocate('RH-1_M-value', .after = 'RH-1-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-2_M-value', .after = 'RH-2-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-3_M-value', .after = 'RH-3-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-4_M-value', .after = 'RH-4-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-5_M-value', .after = 'RH-5-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-6_M-value', .after = 'RH-6-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-7_M-value', .after = 'RH-7-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-8_M-value', .after = 'RH-8-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-10_M-value', .after = 'RH-10-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-11_M-value', .after = 'RH-11-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-12_M-value', .after = 'RH-12-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-14_M-value', .after = 'RH-14-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-15_M-value', .after = 'RH-15-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-16_M-value', .after = 'RH-16-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-18_M-value', .after = 'RH-18-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-19_M-value', .after = 'RH-19-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-21_M-value', .after = 'RH-21-Un')
counts_and_genes <- counts_and_genes %>% relocate('RH-22_M-value', .after = 'RH-22-Un')
counts_and_genes$hgnc_symbol <- NULL
```
```{r}
# Design matrix w a one-way layout

targets$group <- gsub(" ", "", targets$group, fixed = TRUE)
targets$groups <- targets$group 

design <- modelMatrixMeth(model.matrix(~ 0 + groups, data=targets))
design

# Estimate dispersion and model 
y_glm_disp <- estimateDisp(ypr, design=design,  trend="none")
y_glm_fit <- glmFit(y_glm_disp, design)
```
```{r}
# Contrasts of interest for our study 

# Research Question 1 - SMV ischemic vs SMV normal (with SMV normal as reference group)
SMVisch_vs_SMVnormal <- glmLRT(y_glm_fit, contrast = makeContrasts(groupsSMVischemic - groupsSMVnormal, levels=design)) 
#SMVisch_vs_SMVnormal

# Research Question 2 - HSMV ischemic vs HSMV normal (with HSMV normal as reference group)
HSMVisch_vs_HSMVnormal <- glmLRT(y_glm_fit, contrast = makeContrasts(groupsHSMVischemic - groupsHSMVnormal, levels=design)) 
#HSMVisch_vs_HSMVnormal

# Research Question 3 - HSMV ischemic vs HSMV normal vs SMV ischemic vs SMV normal (difference in difference with the difference between 
# SMV ischemic and SMV normal being the reference)
diffisch_vs_diffnormal <- glmLRT(y_glm_fit, contrast = makeContrasts( (groupsHSMVischemic - groupsHSMVnormal) - (groupsSMVischemic - groupsSMVnormal), levels=design))
#diffisch_vs_diffnormal

# Research Question 4 - MVM vs SMV ischemic (with SMV ischemic as reference)
MVM_vs_SMVischemic <- glmLRT(y_glm_fit, contrast = makeContrasts(groupsMVM - groupsSMVischemic, levels=design)) 
#MVM_vs_SMVischemic

# Research Question 5 - HVM vs HSMV ischemic (with HSMV ischemic as reference)
HVM_vs_HSMVischemic <- glmLRT(y_glm_fit, contrast = makeContrasts(groupsHVM - groupsHSMVischemic, levels=design)) 
#HVM_vs_HSMVischemic

# Add corrected p-value columns

# Research Question 1 - SMV normal vs SMV ischemic 
SMVisch_vs_SMVnormal$table$BH <- p.adjust(SMVisch_vs_SMVnormal$table$PValue, method = "BH")
SMVisch_vs_SMVnormal$table$bonferroni <- p.adjust(SMVisch_vs_SMVnormal$table$PValue, method = "bonferroni")

# Research Question 2 - HSMV normal vs HSMV ischemic 
HSMVisch_vs_HSMVnormal$table$BH <- p.adjust(HSMVisch_vs_HSMVnormal$table$PValue, method = "BH")
HSMVisch_vs_HSMVnormal$table$bonferroni <- p.adjust(HSMVisch_vs_HSMVnormal$table$PValue, method = "bonferroni")

# Research Question 3 - SMV normal vs SMV ischemic vs HSMV normal vs HSMV ischemic
diffisch_vs_diffnormal$table$BH <- p.adjust(diffisch_vs_diffnormal$table$PValue, method = "BH")
diffisch_vs_diffnormal$table$bonferroni <- p.adjust(diffisch_vs_diffnormal$table$PValue, method = "bonferroni")

# Research Question 4 - MVM vs SMV ischemic 
MVM_vs_SMVischemic$table$BH <- p.adjust(MVM_vs_SMVischemic$table$PValue, method = "BH")
MVM_vs_SMVischemic$table$bonferroni <- p.adjust(MVM_vs_SMVischemic$table$PValue, method = "bonferroni")

# Research Question 5 - HVM vs HSMV ischemic 
HVM_vs_HSMVischemic$table$BH <- p.adjust(HVM_vs_HSMVischemic$table$PValue, method = "BH")
HVM_vs_HSMVischemic$table$bonferroni <- p.adjust(HVM_vs_HSMVischemic$table$PValue, method = "bonferroni")
```
```{r}
#################
# Make tables 
#################
#get entrez IDs
ss_mart2 <- useMart('ENSEMBL_MART_ENSEMBL', dataset = 'sscrofa_gene_ensembl')

g<-select(ss_mart2, keys=counts_and_genes$ensembl_gene_id, columns=c('entrezgene_id', 'ensembl_gene_id','hgnc_symbol'),
  keytype='ensembl_gene_id', mirror = 'uswest.ensembl.org')

entrez_ids <- aggregate(entrezgene_id ~., g, toString)

dplyr::filter(entrez_ids, ensembl_gene_id == 'ENSSSCG00000017809')

#  hgnc_symbol    ensembl_gene_id        entrezgene_id
#1      GEMIN4 ENSSSCG00000017809 100522827, 100523693

g2 <-select(ss_mart2, keys=counts_and_genes$ensembl_gene_id, columns=c('ensembl_gene_id','entrezgene_accession'),
  keytype='ensembl_gene_id', mirror = 'uswest.ensembl.org')

entrez_accessions <- aggregate(entrezgene_accession ~., g2, toString) 

dplyr::filter(entrez_accessions, ensembl_gene_id == 'ENSSSCG00000017809')

length(unique(entrez_accessions$ensembl_gene_id))
length(unique(entrez_ids$ensembl_gene_id))

entrez_info <- inner_join(entrez_ids, entrez_accessions)

dplyr::filter(entrez_info, ensembl_gene_id == 'ENSSSCG00000017809')

#https://www.ncbi.nlm.nih.gov/gene/100522827
#https://www.ncbi.nlm.nih.gov/gene/100523693

#Note that only https://www.ncbi.nlm.nih.gov/gene/100522827 has the ensembl ID on the descrption page.

entrez_ids <- entrez_info 
entrez_ids$hgnc_symbol <- NULL
entrez_ids$hgnc_symbol <- entrez_ids$entrezgene_accession
entrez_ids$entrezgene_accession <- NULL

counts_and_genes$hgnc_symbol <- NULL
```
```{r}
#need to fix counts and genes table so that situations where counts are 0 are encoded differently/clearly
head(counts_and_genes)

counts_and_genes_long <- pivot_longer(counts_and_genes, cols = -c(ensembl_gene_id))

counts_and_genes_long_sep <- counts_and_genes_long %>% separate(name, c("RH","sample", "measure"), sep = "([-])")


counts_and_genes_long_sep$sample <- str_replace(counts_and_genes_long_sep$sample, '_M', '')
counts_and_genes_long_sep$measure  <- str_replace(counts_and_genes_long_sep$measure , 'value', 'M_value')

counts_and_genes_wide <- pivot_wider(counts_and_genes_long_sep, id_cols = c('ensembl_gene_id', 'RH', 'sample'), names_from = 'measure', values_from = 'value' )


counts_and_genes_wide$M_value_ne <- ifelse(((counts_and_genes_wide$Me==0) & (counts_and_genes_wide$Un==0)), 'not_expressed', counts_and_genes_wide$M_value)

counts_and_genes_wide$M_value <- counts_and_genes_wide$M_value_ne

counts_and_genes_wide$M_value_ne <- NULL

counts_and_genes_wide$original_sample <- paste0(counts_and_genes_wide$RH, '-', counts_and_genes_wide$sample)
counts_and_genes_wide$RH <- NULL
counts_and_genes_wide$sample <- NULL
counts_and_genes_wide$sample <- counts_and_genes_wide$original_sample
counts_and_genes_wide$original_sample <- NULL

counts_and_genes_wider_mvalues <- pivot_wider(counts_and_genes_wide, id_cols = 'ensembl_gene_id', names_from = 'sample', values_from = 'M_value')
counts_and_genes_wider_Me <- pivot_wider(counts_and_genes_wide, id_cols = 'ensembl_gene_id', names_from = 'sample', values_from = 'Me')
counts_and_genes_wider_Un <- pivot_wider(counts_and_genes_wide, id_cols = 'ensembl_gene_id', names_from = 'sample', values_from = 'Un')


counts_and_genes_wider_mvalues <- counts_and_genes_wider_mvalues %>% rename_at(vars(-ensembl_gene_id), paste0, "-M_values")
counts_and_genes_wider_Me <- counts_and_genes_wider_Me %>% rename_at(vars(-ensembl_gene_id), paste0, "-Me")
counts_and_genes_wider_Un <- counts_and_genes_wider_Un %>% rename_at(vars(-ensembl_gene_id), paste0, "-Un")


counts_and_genes_wider <- inner_join(counts_and_genes_wider_Me, counts_and_genes_wider_Un)

counts_and_genes_wider <- inner_join(counts_and_genes_wider, counts_and_genes_wider_mvalues)

nrow(counts_and_genes_wider)

counts_and_genes <- counts_and_genes_wider

write.table(counts_and_genes, 'counts_and_genes.txt', row.names = F, sep = '\t')
```
```{r}
# Question 1 
SMVisch_vs_SMVnormal_results <- data.frame(SMVisch_vs_SMVnormal$table) %>% tibble::rownames_to_column(var = 'ensembl_gene_id')

# add annotations 
SMVisch_vs_SMVnormal_results <- left_join(SMVisch_vs_SMVnormal_results, entrez_ids, by = 'ensembl_gene_id')

# add the counts and M vals
SMVisch_vs_SMVnormal_results <- left_join(SMVisch_vs_SMVnormal_results, counts_and_genes, by = 'ensembl_gene_id')

# filter results based on on nominal p-value
SMVisch_vs_SMVnormal_results_filtered <- SMVisch_vs_SMVnormal_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)


# Question 2 
HSMVisch_vs_HSMVnormal_results <- data.frame(HSMVisch_vs_HSMVnormal$table) %>% tibble::rownames_to_column(var = 'ensembl_gene_id')
# add annotations
HSMVisch_vs_HSMVnormal_results <- left_join(HSMVisch_vs_HSMVnormal_results, entrez_ids, by = 'ensembl_gene_id')
# add the counts and M vals
HSMVisch_vs_HSMVnormal_results <- left_join(HSMVisch_vs_HSMVnormal_results, counts_and_genes, by = 'ensembl_gene_id')

# filter results based on on nominal p-value
HSMVisch_vs_HSMVnormal_results_filtered <- HSMVisch_vs_HSMVnormal_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)

# Question 3 
diffisch_vs_diffnormal_results <- data.frame(diffisch_vs_diffnormal$table) %>% tibble::rownames_to_column(var = 'ensembl_gene_id')
# add annotations
diffisch_vs_diffnormal_results <- left_join(diffisch_vs_diffnormal_results, entrez_ids, by = 'ensembl_gene_id')
# add the counts and M vals
diffisch_vs_diffnormal_results <- left_join(diffisch_vs_diffnormal_results, counts_and_genes, by = 'ensembl_gene_id')

# filter results based on on nominal p-value
diffisch_vs_diffnormal_results_filtered <- diffisch_vs_diffnormal_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)

# Question 4 
MVM_vs_SMVischemic_results <- data.frame(MVM_vs_SMVischemic$table) %>% tibble::rownames_to_column(var = 'ensembl_gene_id')
# add annotations
MVM_vs_SMVischemic_results <- left_join(MVM_vs_SMVischemic_results, entrez_ids, by = 'ensembl_gene_id')
# add the counts and M vals
MVM_vs_SMVischemic_results <- left_join(MVM_vs_SMVischemic_results, counts_and_genes, by = 'ensembl_gene_id')

# filter results based on on nominal p-value
MVM_vs_SMVischemic_results_filtered <- MVM_vs_SMVischemic_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)

# Question 5 
HVM_vs_HSMVischemic_results <- data.frame(HVM_vs_HSMVischemic$table) %>% tibble::rownames_to_column(var = 'ensembl_gene_id')
# add annotations
HVM_vs_HSMVischemic_results <- left_join(HVM_vs_HSMVischemic_results, entrez_ids, by = 'ensembl_gene_id')
# add the counts and M vals
HVM_vs_HSMVischemic_results <- left_join(HVM_vs_HSMVischemic_results, counts_and_genes, by = 'ensembl_gene_id')

# filter results based on on nominal p-value
HVM_vs_HSMVischemic_results_filtered <- HVM_vs_HSMVischemic_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)

save.image('/home/rstudio/dat/session7.Rdata')

```

```{r}
# This code chunk is from the `List_overlap.R` script that Jordan made.
library(dplyr)

shared_loci <- dplyr::inner_join(HSMVisch_vs_HSMVnormal_results_filtered, SMVisch_vs_SMVnormal_results_filtered, by="ensembl_gene_id", keep = F) %>% 
  dplyr::rename_at(vars(ends_with(".x")),~str_replace(., "\\..$","")) %>% 
  dplyr::select_at(vars(-ends_with(".y")))

head(shared_loci)


# Effect in HSMV but not in SMV
HSMV_only <- dplyr::anti_join(HSMVisch_vs_HSMVnormal_results_filtered, SMVisch_vs_SMVnormal_results_filtered, by="ensembl_gene_id")


# Effect in SMV but not in HSMV 
SMV_only <- dplyr::anti_join(SMVisch_vs_SMVnormal_results_filtered, HSMVisch_vs_HSMVnormal_results_filtered, by="ensembl_gene_id")


# Write excel files
#write.xlsx(shared_loci, file = '/home/rstudio/dat/both_HSMV_and_SMV.xlsx')
#write.xlsx(HSMV_only, file = '/home/rstudio/dat/HSMV_effect_only.xlsx')
#write.xlsx(SMV_only, file = '/home/rstudio/dat/SMV_effect_only.xlsx')
```


```{r}
#here's an example of how to make heatmaps from the normalized data tables:
load('/home/rstudio/dat/session7.Rdata')


library(ggplot2)
library(viridis)
library(stringr)
labels <- targets
```


```{r}
#pivot longer and fix columns so that we can inner join the sample information:
SMVisch_vs_SMVnormal_results_filtered_long <- SMVisch_vs_SMVnormal_results_filtered %>% tidyr::pivot_longer(cols = starts_with('RH-'), names_to = "measurement", values_to = "value", values_transform = list(value = as.character))
SMVisch_vs_SMVnormal_results_filtered_long <- tidyr::separate(SMVisch_vs_SMVnormal_results_filtered_long, measurement, into = c('val1', 'sample', 'measurements'), sep = '-')
SMVisch_vs_SMVnormal_results_filtered_long$val1 <- NULL

SMVisch_vs_SMVnormal_results_filtered_long$sample <- paste0('RH-', SMVisch_vs_SMVnormal_results_filtered_long$sample)
SMVisch_vs_SMVnormal_results_filtered_long <- dplyr::inner_join(SMVisch_vs_SMVnormal_results_filtered_long, labels, by = 'sample')

#this bit sets the order of the ensembl_gene_id and hgnc_symbol to be sorted by logFC
SMVisch_vs_SMVnormal_results_filtered_long$ensembl_gene_id <- 
  factor(SMVisch_vs_SMVnormal_results_filtered_long$ensembl_gene_id, 
         levels = 
           (unique(SMVisch_vs_SMVnormal_results_filtered_long$ensembl_gene_id)[
             order(SMVisch_vs_SMVnormal_results_filtered_long$logFC, method = 'radix')
             ]))
SMVisch_vs_SMVnormal_results_filtered_long$hgnc_symbol <- 
  factor(SMVisch_vs_SMVnormal_results_filtered_long$hgnc_symbol, 
         levels = 
           (unique(SMVisch_vs_SMVnormal_results_filtered_long$hgnc_symbol)[
             order(SMVisch_vs_SMVnormal_results_filtered_long$logFC, method = 'radix')
             ]))

#this bit in the plotting code gets rid of NA in the hgnc symbols: SMVisch_vs_SMVnormal_results_filtered_long[!is.na(SMVisch_vs_SMVnormal_results_filtered_long$hgnc_symbol), ]
```

```{r}
#then we plot

#Here we can note the genes in the middle tend to have M-values very close to 0.

ggplot(data = 
         SMVisch_vs_SMVnormal_results_filtered_long[!is.na(SMVisch_vs_SMVnormal_results_filtered_long$hgnc_symbol), ] %>% 
         dplyr::filter(., group == "SMVischemic" | group == "SMVnormal") %>% 
         dplyr::filter(., measurements == 'M_values'),
       mapping = aes(
         x = sample, 
         y = hgnc_symbol, 
         fill = as.numeric(value))) +
  geom_tile() + 
  scale_fill_distiller(palette = "RdYlBu") +
  facet_grid(~group, scales = 'free') +
  labs(title="M values -- log2( (Me + .01) / (Un + .01) )",
        x ="Samples", y = "Gene Symbol", fill = "M Value")


#Here we can note the genes in the middle tend to have high methylated counts, showing the trend where genes that have high counts are still counted as statistically significant although the fold change is subtle and small.
ggplot(data = 
         SMVisch_vs_SMVnormal_results_filtered_long[!is.na(SMVisch_vs_SMVnormal_results_filtered_long$hgnc_symbol), ] %>% 
         dplyr::filter(., group == "SMVischemic" | group == "SMVnormal") %>% 
         dplyr::filter(., measurements == 'Me'),
       mapping = aes(
         x = sample, 
         y = hgnc_symbol, 
         fill = as.numeric(value))) +
  geom_tile() + 
  scale_fill_distiller(palette = "RdYlBu") +
  facet_grid(~group, scales = 'free') +
  labs(title="Log transformed Methylation Values",
        x ="Samples", y = "Gene Symbol", fill = "Log(Methylated Counts)")





#save.image('/home/rstudio/dat/session8.Rdata')



```


```{r}

HSMVisch_vs_HSMVnormal_results_filtered_long <- as.data.frame(HSMVisch_vs_HSMVnormal_results_filtered) %>% tidyr::pivot_longer(cols = starts_with('R'), names_to = "measurement", values_to = "value") 
HSMVisch_vs_HSMVnormal_results_filtered_long <- separate(HSMVisch_vs_HSMVnormal_results_filtered_long, measurement, into = c('val1', 'val2'), sep = 3)
HSMVisch_vs_HSMVnormal_results_filtered_long$val1 <- NULL
measurements <- gl(3,1,nrow(HSMVisch_vs_HSMVnormal_results_filtered_long), labels=c("Me_counts", "Un_counts", "M_value"))
HSMVisch_vs_HSMVnormal_results_filtered_long$measurement <- measurements
HSMVisch_vs_HSMVnormal_results_filtered_long <- separate(HSMVisch_vs_HSMVnormal_results_filtered_long, val2, into = c('sample', 'val3'), sep = '-')
HSMVisch_vs_HSMVnormal_results_filtered_long$sample <- str_replace(HSMVisch_vs_HSMVnormal_results_filtered_long$sample, '_M', '')
HSMVisch_vs_HSMVnormal_results_filtered_long$val3 <- str_replace(HSMVisch_vs_HSMVnormal_results_filtered_long$val3, 'value', 'M_value')
HSMVisch_vs_HSMVnormal_results_filtered_long$measurement <- HSMVisch_vs_HSMVnormal_results_filtered_long$val3
HSMVisch_vs_HSMVnormal_results_filtered_long$val3 <- NULL
HSMVisch_vs_HSMVnormal_results_filtered_long$sample <- paste0('RH-',HSMVisch_vs_HSMVnormal_results_filtered_long$sample)
HSMVisch_vs_HSMVnormal_results_filtered_long <- pivot_wider(HSMVisch_vs_HSMVnormal_results_filtered_long, names_from = 'measurement', values_from = 'value')
HSMVisch_vs_HSMVnormal_results_filtered_long <- inner_join(HSMVisch_vs_HSMVnormal_results_filtered_long, labels, by = 'sample')
HSMVisch_vs_HSMVnormal_results_filtered_long$M_value <- as.numeric(str_trim(HSMVisch_vs_HSMVnormal_results_filtered_long$M_value, side = c("both")))
HSMVisch_vs_HSMVnormal_results_filtered_long$Me <- as.numeric(str_trim(HSMVisch_vs_HSMVnormal_results_filtered_long$Me, side = c("both")))
HSMVisch_vs_HSMVnormal_results_filtered_long$Un <- as.numeric(str_trim(HSMVisch_vs_HSMVnormal_results_filtered_long$Un, side = c("both")))

diffisch_vs_diffnormal_results_filtered_long <- as.data.frame(diffisch_vs_diffnormal_results_filtered) %>% tidyr::pivot_longer(cols = starts_with('R'), names_to = "measurement", values_to = "value") 
diffisch_vs_diffnormal_results_filtered_long <- separate(diffisch_vs_diffnormal_results_filtered_long, measurement, into = c('val1', 'val2'), sep = 3)
diffisch_vs_diffnormal_results_filtered_long$val1 <- NULL
measurements <- gl(3,1,nrow(diffisch_vs_diffnormal_results_filtered_long), labels=c("Me_counts", "Un_counts", "M_value"))
diffisch_vs_diffnormal_results_filtered_long$measurement <- measurements
diffisch_vs_diffnormal_results_filtered_long <- separate(diffisch_vs_diffnormal_results_filtered_long, val2, into = c('sample', 'val3'), sep = '-')
diffisch_vs_diffnormal_results_filtered_long$sample <- str_replace(diffisch_vs_diffnormal_results_filtered_long$sample, '_M', '')
diffisch_vs_diffnormal_results_filtered_long$val3 <- str_replace(diffisch_vs_diffnormal_results_filtered_long$val3, 'value', 'M_value')
diffisch_vs_diffnormal_results_filtered_long$measurement <- diffisch_vs_diffnormal_results_filtered_long$val3
diffisch_vs_diffnormal_results_filtered_long$val3 <- NULL
diffisch_vs_diffnormal_results_filtered_long$sample <- paste0('RH-',diffisch_vs_diffnormal_results_filtered_long$sample)
diffisch_vs_diffnormal_results_filtered_long <- pivot_wider(diffisch_vs_diffnormal_results_filtered_long, names_from = 'measurement', values_from = 'value')
diffisch_vs_diffnormal_results_filtered_long <- inner_join(diffisch_vs_diffnormal_results_filtered_long, labels, by = 'sample')
diffisch_vs_diffnormal_results_filtered_long$M_value <- as.numeric(str_trim(diffisch_vs_diffnormal_results_filtered_long$M_value, side = c("both")))
diffisch_vs_diffnormal_results_filtered_long$Me <- as.numeric(str_trim(diffisch_vs_diffnormal_results_filtered_long$Me, side = c("both")))
diffisch_vs_diffnormal_results_filtered_long$Un <- as.numeric(str_trim(diffisch_vs_diffnormal_results_filtered_long$Un, side = c("both")))

MVM_vs_SMVischemic_results_filtered_long <- as.data.frame(MVM_vs_SMVischemic_results_filtered) %>% tidyr::pivot_longer(cols = starts_with('R'), names_to = "measurement", values_to = "value") 
MVM_vs_SMVischemic_results_filtered_long <- separate(MVM_vs_SMVischemic_results_filtered_long, measurement, into = c('val1', 'val2'), sep = 3)
MVM_vs_SMVischemic_results_filtered_long$val1 <- NULL
measurements <- gl(3,1,nrow(MVM_vs_SMVischemic_results_filtered_long), labels=c("Me_counts", "Un_counts", "M_value"))
MVM_vs_SMVischemic_results_filtered_long$measurement <- measurements
MVM_vs_SMVischemic_results_filtered_long <- separate(MVM_vs_SMVischemic_results_filtered_long, val2, into = c('sample', 'val3'), sep = '-')
MVM_vs_SMVischemic_results_filtered_long$sample <- str_replace(MVM_vs_SMVischemic_results_filtered_long$sample, '_M', '')
MVM_vs_SMVischemic_results_filtered_long$val3 <- str_replace(MVM_vs_SMVischemic_results_filtered_long$val3, 'value', 'M_value')
MVM_vs_SMVischemic_results_filtered_long$measurement <- MVM_vs_SMVischemic_results_filtered_long$val3
MVM_vs_SMVischemic_results_filtered_long$val3 <- NULL
MVM_vs_SMVischemic_results_filtered_long$sample <- paste0('RH-',MVM_vs_SMVischemic_results_filtered_long$sample)
MVM_vs_SMVischemic_results_filtered_long <- pivot_wider(MVM_vs_SMVischemic_results_filtered_long, names_from = 'measurement', values_from = 'value')
MVM_vs_SMVischemic_results_filtered_long <- inner_join(MVM_vs_SMVischemic_results_filtered_long, labels, by = 'sample')
MVM_vs_SMVischemic_results_filtered_long$M_value <- as.numeric(str_trim(MVM_vs_SMVischemic_results_filtered_long$M_value, side = c("both")))
MVM_vs_SMVischemic_results_filtered_long$Me <- as.numeric(str_trim(MVM_vs_SMVischemic_results_filtered_long$Me, side = c("both")))
MVM_vs_SMVischemic_results_filtered_long$Un <- as.numeric(str_trim(MVM_vs_SMVischemic_results_filtered_long$Un, side = c("both")))

HVM_vs_HSMVischemic_results_filtered_long <- as.data.frame(HVM_vs_HSMVischemic_results_filtered) %>% tidyr::pivot_longer(cols = starts_with('R'), names_to = "measurement", values_to = "value") 
HVM_vs_HSMVischemic_results_filtered_long <- separate(HVM_vs_HSMVischemic_results_filtered_long, measurement, into = c('val1', 'val2'), sep = 3)
HVM_vs_HSMVischemic_results_filtered_long$val1 <- NULL
measurements <- gl(3,1,nrow(HVM_vs_HSMVischemic_results_filtered_long), labels=c("Me_counts", "Un_counts", "M_value"))
HVM_vs_HSMVischemic_results_filtered_long$measurement <- measurements
HVM_vs_HSMVischemic_results_filtered_long <- separate(HVM_vs_HSMVischemic_results_filtered_long, val2, into = c('sample', 'val3'), sep = '-')
HVM_vs_HSMVischemic_results_filtered_long$sample <- str_replace(HVM_vs_HSMVischemic_results_filtered_long$sample, '_M', '')
HVM_vs_HSMVischemic_results_filtered_long$val3 <- str_replace(HVM_vs_HSMVischemic_results_filtered_long$val3, 'value', 'M_value')
HVM_vs_HSMVischemic_results_filtered_long$measurement <- HVM_vs_HSMVischemic_results_filtered_long$val3
HVM_vs_HSMVischemic_results_filtered_long$val3 <- NULL
HVM_vs_HSMVischemic_results_filtered_long$sample <- paste0('RH-',HVM_vs_HSMVischemic_results_filtered_long$sample)
HVM_vs_HSMVischemic_results_filtered_long <- pivot_wider(HVM_vs_HSMVischemic_results_filtered_long, names_from = 'measurement', values_from = 'value')
HVM_vs_HSMVischemic_results_filtered_long <- inner_join(HVM_vs_HSMVischemic_results_filtered_long, labels, by = 'sample')
HVM_vs_HSMVischemic_results_filtered_long$M_value <- as.numeric(str_trim(HVM_vs_HSMVischemic_results_filtered_long$M_value, side = c("both")))
HVM_vs_HSMVischemic_results_filtered_long$Me <- as.numeric(str_trim(HVM_vs_HSMVischemic_results_filtered_long$Me, side = c("both")))
HVM_vs_HSMVischemic_results_filtered_long$Un <- as.numeric(str_trim(HVM_vs_HSMVischemic_results_filtered_long$Un, side = c("both")))

```


```{r}

```

```{r}

#this bit sets the order of the ensembl_gene_id and hgnc_symbol to be sorted by logFC
HSMVisch_vs_HSMVnormal_results_filtered_long$ensembl_gene_id <- 
  factor(HSMVisch_vs_HSMVnormal_results_filtered_long$ensembl_gene_id, 
         levels = 
           (unique(HSMVisch_vs_HSMVnormal_results_filtered_long$ensembl_gene_id)[
             order(HSMVisch_vs_HSMVnormal_results_filtered_long$logFC, method = 'radix')
             ]))

HSMVisch_vs_HSMVnormal_results_filtered_long$hgnc_symbol <- 
  factor(HSMVisch_vs_HSMVnormal_results_filtered_long$hgnc_symbol, 
         levels = 
           (unique(HSMVisch_vs_HSMVnormal_results_filtered_long$hgnc_symbol)[
             order(HSMVisch_vs_HSMVnormal_results_filtered_long$logFC, method = 'radix')
             ]))

#this bit sets the order of the ensembl_gene_id and hgnc_symbol to be sorted by logFC
diffisch_vs_diffnormal_results_filtered_long$ensembl_gene_id <- 
  factor(diffisch_vs_diffnormal_results_filtered_long$ensembl_gene_id, 
         levels = 
           (unique(diffisch_vs_diffnormal_results_filtered_long$ensembl_gene_id)[
             order(diffisch_vs_diffnormal_results_filtered_long$logFC, method = 'radix')
             ]))

diffisch_vs_diffnormal_results_filtered_long$hgnc_symbol <- 
  factor(diffisch_vs_diffnormal_results_filtered_long$hgnc_symbol, 
         levels = 
           (unique(diffisch_vs_diffnormal_results_filtered_long$hgnc_symbol)[
             order(diffisch_vs_diffnormal_results_filtered_long$logFC, method = 'radix')
             ]))

#this bit sets the order of the ensembl_gene_id and hgnc_symbol to be sorted by logFC
MVM_vs_SMVischemic_results_filtered_long$ensembl_gene_id <- 
  factor(MVM_vs_SMVischemic_results_filtered_long$ensembl_gene_id, 
         levels = 
           (unique(MVM_vs_SMVischemic_results_filtered_long$ensembl_gene_id)[
             order(MVM_vs_SMVischemic_results_filtered_long$logFC, method = 'radix')
             ]))

MVM_vs_SMVischemic_results_filtered_long$hgnc_symbol <- 
  factor(MVM_vs_SMVischemic_results_filtered_long$hgnc_symbol, 
         levels = 
           (unique(MVM_vs_SMVischemic_results_filtered_long$hgnc_symbol)[
             order(MVM_vs_SMVischemic_results_filtered_long$logFC, method = 'radix')
             ]))

#this bit sets the order of the ensembl_gene_id and hgnc_symbol to be sorted by logFC
HVM_vs_HSMVischemic_results_filtered_long$ensembl_gene_id <- 
  factor(HVM_vs_HSMVischemic_results_filtered_long$ensembl_gene_id, 
         levels = 
           (unique(HVM_vs_HSMVischemic_results_filtered_long$ensembl_gene_id)[
             order(HVM_vs_HSMVischemic_results_filtered_long$logFC, method = 'radix')
             ]))

HVM_vs_HSMVischemic_results_filtered_long$hgnc_symbol <- 
  factor(HVM_vs_HSMVischemic_results_filtered_long$hgnc_symbol, 
         levels = 
           (unique(HVM_vs_HSMVischemic_results_filtered_long$hgnc_symbol)[
             order(HVM_vs_HSMVischemic_results_filtered_long$logFC, method = 'radix')
             ]))
```


```{r}

```


```{r}
ggplot(data = (HSMVisch_vs_HSMVnormal_results_filtered_long %>% filter(group == "HSMVischemic" | group == "HSMVnormal")),
       mapping = aes(
         x = sample, 
         y = hgnc_symbol, 
         fill = M_value)) +
  geom_tile() + 
  scale_fill_distiller(palette = "RdYlBu") +
  facet_grid(~group, scales = 'free')

ggsave('/home/rstudio/dat/HSMVisch_vs_HSMVnormal_results_filtered_heatmap.png', device = 'png', height = 12)


ggplot(data = (diffisch_vs_diffnormal_results_filtered_long %>% filter(group == "SMVischemic" | group == "SMVnormal" | group == "HSMVischemic" | group == "HSMVnormal")),
       mapping = aes(
         x = sample, 
         y = hgnc_symbol, 
         fill = M_value)) +
  geom_tile() + 
  scale_fill_distiller(palette = "RdYlBu") +
  facet_grid(~group, scales = 'free')

ggsave('/home/rstudio/dat/diffisch_vs_diffnormal_results_filtered_heatmap.png', device = 'png', height = 12)


ggplot(data = (MVM_vs_SMVischemic_results_filtered_long %>% filter(group == "MVM" | group == "SMVischemic")),
       mapping = aes(
         x = sample, 
         y = hgnc_symbol, 
         fill = M_value)) +
  geom_tile() + 
  scale_fill_distiller(palette = "RdYlBu") +
  facet_grid(~group, scales = 'free')

ggsave('/home/rstudio/dat/MVM_vs_SMVischemic_results_filtered_heatmap.png', device = 'png', height = 12)


ggplot(data = (HVM_vs_HSMVischemic_results_filtered_long %>% filter(group == "HVM" | group == "HSMVischemic")),
       mapping = aes(
         x = sample, 
         y = hgnc_symbol, 
         fill = M_value)) +
  geom_tile() + 
  scale_fill_distiller(palette = "RdYlBu") +
  facet_grid(~group, scales = 'free')

ggsave('/home/rstudio/dat/HVM_vs_HSMVischemic_results_filtered_heatmap.png', device = 'png', height = 12)



```

```{r}
#First, make the readme tab:
readme_promoters <- "Methylation data was filtered so that each loci had to have at least 5 counts in at least 3 samples and we also removed loci where the samples were never methylated or always methylated, resulting in 84999 loci. "
readme_dtn <- "The distanceToNearest function from the GRanges R package was used to determine how close each loci was to a transcription start site. Loci that were less than 2kb from a transcription start site were grouped together and their counts summed. The summed counts for all loci within 2kb of a transcription start site were used as the input for edgeR."
readme_glm <- "glmFit, glmLRT, and makeContrasts functions in edgeR were used to fit a genewise negative binomial GLM and make comparisons of interest based on the research question."
readme_ensembl_gene_id <- "`ensembl_gene_id` is the Ensembl gene ID associated with each TSS"
readme_logFC <- "`logFC` indicates the log2-fold change of expression between the two conditions tested. For `SMVisch_vs_SMVnormal_results_filtered` results, positive values indicate higher methylation rates in SMVnormal relative to SMVisch, negative values indicate lower methylation rates in SMVnormal vs SMVisch. For `HSMVisch_vs_HSMVnormal_results_filtered` results, positive values indicate higher methylation rates in HSMVnormal relative to HSMVisch, negative values indicate lower methylation rates in HSMVnormal relative to HSMVisch. For `diffisch_vs_diffnormal_results_filtered` results, positive values indicate ...? For `MVM_vs_SMVischemic_results_filtered` results, positive values indicate higher methylation rates in MVM relative to SMVischemic, negative values indicate lower methylation rates in MVM relative to SMVischemic."
readme_logCPM <- "`logCPM` is the average log2-counts per million, the average taken over all libraries used in the experiment."
readme_LR <- "`LR` is the likelihood ratio statistics (larger LR, smaller p-value"
readme_PValue <- "`PValue` probability of obtaining results at least as extreme as the results actually observed, under the assumption that the null hypothesis is correct. The 'filtered' tab are filtered based on a nominal P-value cutoff of 0.01."
readme_BH <- "`BH` is Benjamini & Hochberg corrected p-values"
readme_bonferroni <- "`bonferroni` Bonferroni corrected p-values"
readme_entrezgene_id <- "`entrezgene_id` is the NCBI/Entrez gene ID associated with each TSS (if available)"
readme_hgnc_symbol <- "`hgnc_symbol` is the HGNC symbol associated with each TSS (if available)"
readme_MeUn <- "The columns with the format `RH-#-Me` or `RH-#-Un` indicate the total methylated or unmethylated counts within 2KB of the transcription start site for a given sample and gene."
readme_MVal <- "The columns with the format `RH-#_M-value` indicate the M-values for a given sample and gene. The M-values were calculated as follows: log2( (Me + .01) / (Un + .01) )"

readme_sheet <- rbind(readme_promoters, readme_dtn, readme_glm, readme_ensembl_gene_id, readme_logFC, readme_logCPM, readme_LR, readme_PValue, readme_BH, readme_bonferroni, readme_entrezgene_id, readme_hgnc_symbol, readme_MeUn, readme_MVal)


#write.xlsx(shared_loci, file = '/home/rstudio/dat/both_HSMV_and_SMV.xlsx')
#write.xlsx(HSMV_only, file = '/home/rstudio/dat/HSMV_effect_only.xlsx')
#write.xlsx(SMV_only, file = '/home/rstudio/dat/SMV_effect_only.xlsx')

#write outputs
list_1 <- list(readme_sheet, SMVisch_vs_SMVnormal_results, SMVisch_vs_SMVnormal_results_filtered)
names(list_1) <- c('readme', 'SMVisch_vs_SMVnormal', 'SMVisch_vs_SMVnormal_filtered')
write.xlsx(list_1, file = '/home/rstudio/dat/SMVisch_vs_SMVnormal_results.xlsx')

list_2 <- list(readme_sheet, HSMVisch_vs_HSMVnormal_results, HSMVisch_vs_HSMVnormal_results_filtered)
names(list_2) <- c('readme', 'HSMVisch_vs_HSMVnormal', 'HSMVisch_vs_HSMVnormal_filtered')
write.xlsx(list_2, file = '/home/rstudio/dat/HSMVisch_vs_HSMVnormal_results.xlsx')

list_3 <- list(readme_sheet, diffisch_vs_diffnormal_results, diffisch_vs_diffnormal_results_filtered)
names(list_3) <- c('readme', 'diffisch_vs_diffnormal', 'diffisch_vs_diffnormal_filtered')
write.xlsx(list_3, file = '/home/rstudio/dat/diffisch_vs_diffnormal_results.xlsx')

list_4 <- list(readme_sheet, MVM_vs_SMVischemic_results, MVM_vs_SMVischemic_results_filtered)
names(list_4) <- c('readme', 'MVM_vs_SMVischemic', 'MVM_vs_SMVischemic_filtered')
write.xlsx(list_4, file = '/home/rstudio/dat/MVM_vs_SMVischemic_results.xlsx')

list_5 <- list(readme_sheet, HVM_vs_HSMVischemic_results, HVM_vs_HSMVischemic_results_filtered)
names(list_5) <- c('readme', 'HVM_vs_HSMVischemic', 'HVM_vs_HSMVischemic_filtered')
write.xlsx(list_5, file = '/home/rstudio/dat/HVM_vs_HSMVischemic_results.xlsx')

```

```{r}
nrow(SMVisch_vs_SMVnormal_results_filtered) #132
nrow(HSMVisch_vs_HSMVnormal_results_filtered) #131
nrow(diffisch_vs_diffnormal_results_filtered) #154
nrow(MVM_vs_SMVischemic_results_filtered) #194
nrow(HVM_vs_HSMVischemic_results_filtered) #155
```

