---
title: "R_rrbs_analysis_latest"
author: "Joselynn"
date: "7/22/2021"
output: html_document
---

```{r, include = FALSE}
library("edgeR")
library("bsseq")
library("dplyr")
library("tidyr")
library("openxlsx")
library("biomaRt")
library("dplyr")
library("GenomicRanges")
```

```{r}
targets <- read.table('/home/rstudio/dat/targets.txt', header = T, sep = '\t')
files <- targets$file             
yall <- readBismark2DGE(files, sample.names=targets$sample)

save.image('/home/rstudio/dat/session1.Rdata')
```

```{r}
load('/home/rstudio/dat/session1.Rdata')


yall <- yall[yall$genes[,'Chr'] == 1 | yall$genes[,'Chr'] == 2| yall$genes[,'Chr'] == 3| yall$genes[,'Chr'] == 4| yall$genes[,'Chr'] == 5| yall$genes[,'Chr'] == 6| yall$genes[,'Chr'] == 7| yall$genes[,'Chr'] == 8| yall$genes[,'Chr'] == 9| yall$genes[,'Chr'] == 10| yall$genes[,'Chr'] == 11| yall$genes[,'Chr'] == 12| yall$genes[,'Chr'] == 13| yall$genes[,'Chr'] == 14| yall$genes[,'Chr'] == 15| yall$genes[,'Chr'] == 16| yall$genes[,'Chr'] == 17| yall$genes[,'Chr'] == 18| yall$genes[,'Chr'] == 'X'| yall$genes[,'Chr'] == 'Y',]

sort_order <- c(1:18, 'X', 'Y')

yall$genes$Chr <- factor(yall$genes$Chr, levels=sort_order)

o <- order(yall$genes$Chr, yall$genes$Locus)

yall <- yall[o,]        

table(yall$genes$Chr)

#      1       2       3       4       5       6       7       8       9      10      11      12      13      14      15      16      17      18       X 
#2351408 1881377 2159242 1592211 1349283 2801759 1660304 1306871 1540861 1060010 1051247 1440491 1592620 1736525 1196939  788249 1000426  918874  872986 
#      Y 
#  43712 

save.image('/home/rstudio/dat/session2.Rdata')
```


```{r}
load('/home/rstudio/dat/session2.Rdata')

# make factor levels of Me and Un repeated along the length of the columns 
Methylation <- gl(2,1,ncol(yall), labels=c("Me","Un"))
Me <- yall$counts[, Methylation=="Me"]
Un <- yall$counts[, Methylation=="Un"]

# combine methylated and unmethylated reads to get total coverage
Coverage <- Me + Un

# Get a list of CpGs with at least 5 counts (methylated and unmethylated) in 3 samples 
HasCoverage <- rowSums(Coverage >= 5) == 3

# List of CpGs that are not 0 both
# this filters out CpG that are never methylated or always methylated as this provides no info about differential methylation 
HasBoth <- rowSums(Me) > 0 & rowSums(Un) > 0 

# Check sample size after filtering requirements are met 
table(HasCoverage, HasBoth)

save.image('/home/rstudio/dat/session3.Rdata')
```

#        HasBoth
#HasCoverage    FALSE     TRUE
#      FALSE 17958491 10294081
#      TRUE      7824    84999
#84999 have both

```{r}
load('/home/rstudio/dat/session3.Rdata')

# apply those coverage filters
y <- yall[HasCoverage & HasBoth,, keep.lib.sizes=FALSE] 

# assign library sizes to each sample to be a sum of methylated and unmethylated counts
TotalLibSize <- y$samples$lib.size[Methylation=="Me"] + y$samples$lib.size[Methylation=="Un"]
y$samples$lib.size <- rep(TotalLibSize, each=2)
y$samples

save.image('/home/rstudio/dat/session4.Rdata')
table(y$genes$Chr)

#   1     2     3     4     5     6     7     8     9    10    11    12    13 
# 5275  6450  8154  4564  4175 10654  4802  3292  4270  3174  3831  6439  3177 
#   14    15    16    17    18     X     Y 
# 4443  2939  1952  3530  3203   524   151 
```

```{r}
load('/home/rstudio/dat/session4.Rdata')

#get all the genes used in  the DML analysis..
y_loc <- as.data.frame(y$genes)
y_gr <- makeGRangesFromDataFrame(y_loc,ignore.strand = T, start.field = 'Locus', end.field = 'Locus', seqnames.field = 'Chr')

#get granges for all tss in genome..

ss_mart <- useMart("ensembl", dataset = "sscrofa_gene_ensembl")

#attributes <- listAttributes(ss_mart)
#attributes[grep("symbol", attributes$description, ignore.case = TRUE), ]
#attributes[grep("transcription", attributes$description, ignore.case = TRUE), ]
#attributes[grep("vgnc", attributes$description, ignore.case = TRUE), ]

ss_tss <- getBM(attributes = c("transcription_start_site", 
                               "chromosome_name", 
                               "start_position",
                               "end_position",
                               "ensembl_gene_id", 
                               "strand", 
                               "uniprot_gn_symbol"), 
                mart = ss_mart)

ss_tss_2 <- getBM(attributes = c("transcription_start_site", 
                                 "chromosome_name", 
                                 "start_position",
                                 "end_position",
                                 "hgnc_symbol",
                                 "entrezgene_id",
                                 "description"), 
                  mart = ss_mart)

ss_tss <- as.data.frame(ss_tss)
ss_tss_2 <- as.data.frame(ss_tss_2)
ss_tss_all <- inner_join(ss_tss, ss_tss_2, by = c("transcription_start_site", "chromosome_name", "start_position", "end_position"))
ss_tss <- ss_tss_all

#find and replace the 'strand' column so that -1 is - and 1 is +
library(stringr)
ss_tss$strand <- str_replace(string = ss_tss$strand, pattern = '-1', replacement = '-')
ss_tss$strand <- str_replace(string = ss_tss$strand, pattern = '1', replacement = '+')

#filter so we only have chromosomes
ss_tss_filt <- dplyr::filter(ss_tss, chromosome_name=='1' |  chromosome_name=='2' |chromosome_name=='3' |chromosome_name=='4' |chromosome_name=='5' |chromosome_name=='6' |chromosome_name=='7' |chromosome_name=='8' |chromosome_name=='9' |chromosome_name=='10' |chromosome_name=='11' |chromosome_name=='12' |chromosome_name=='13' |chromosome_name=='14' |chromosome_name=='15' |chromosome_name=='16' |chromosome_name=='17' |chromosome_name=='18' |chromosome_name=='X' |chromosome_name=='Y')
ss_tss <- ss_tss_filt

# turn to granges
ss_tss_gr <- makeGRangesFromDataFrame(ss_tss, start.field = 'start_position', end.field = 'end_position', strand.field = 'strand')

# add metadata
ss_tss_gr$transcription_start_site <- ss_tss$transcription_start_site
ss_tss_gr$ensembl_gene_id <- ss_tss$ensembl_gene_id
ss_tss_gr$uniprot_gn_symbol <- ss_tss$symbol
ss_tss_gr$embl <- ss_tss$embl
ss_tss_gr$hgnc_symbol <- ss_tss$hgnc_symbol
ss_tss_gr$uniprot_gn_symbol <- ss_tss$uniprot_gn_symbol
ss_tss_gr$entrezgene_id <- ss_tss$entrezgene_id

save.image('/home/rstudio/dat/session5.Rdata')
```

```{r}
load('/home/rstudio/dat/session5.Rdata')

#distancetonearest to figure out how far away from a TSS each locus is (use this later to group into promoters)
dtn <- as.data.frame(distanceToNearest(x = y_gr, subject = ss_tss_gr))

#slice to get the subject/tss hits:
tss_slice <- as.data.frame(ss_tss_gr) %>% slice(dtn$subjectHits)

#add the 'distance' column to tss_slice:
tss_slice$distance <- dtn$distance

#add all the tss info to the rrbs data:
y$genes$nearest_gene_start <- tss_slice$start
y$genes$nearest_gene_end <- tss_slice$end
y$genes$nearest_gene_strand <- tss_slice$strand
y$genes$nearest_transcription_start_site <- tss_slice$transcription_start_site
y$genes$distance_to_nearest_tss <- tss_slice$distance
y$genes$ensembl_gene_id <- tss_slice$ensembl_gene_id
y$genes$hgnc_symbol <- tss_slice$hgnc_symbol
y$genes$entrezgene_id <- tss_slice$entrezgene_id

#get loci that are in promoters (e.g., less than 2kb away from TSS)
InPromoter <- y$genes$distance_to_nearest_tss <= 2000

#subset to only get loci in promoters
yIP <- y[InPromoter,,keep.lib.sizes=FALSE]

#get total counts for each promoter region
ypr <- rowsum(yIP, yIP$genes$ensembl_gene_id, reorder=FALSE)

#re-set library sizes based to be the average of the total read counts for the methylated and unmethylated libraries.

TotalLibSizepr <- 0.5*ypr$samples$lib.size[Methylation=="Me"] + 0.5*ypr$samples$lib.size[Methylation=="Un"]
ypr$samples$lib.size <- rep(TotalLibSizepr, each=2)
ypr$samples

Me <- ypr$counts[, Methylation=="Me"]
Un <- ypr$counts[, Methylation=="Un"]

M01 <- data.frame(log2((Me + .01) / (Un + .01)))
M01$ensembl_gene_id <- rownames(M01)


save.image('/home/rstudio/dat/session6.Rdata')
```
```{r}
#In microarray methylation studies, a common measure of methylation level is the M-value, which is defined as M = log 2 {(Me + α) /(Un + α)} where Me and Un are the methylated and unmethylated intensities and α is some suitable offset to avoid taking logarithms of zero 39. The M-value can be interpreted as the base2 logit transformation of the proportion of methylated signal at each locus. (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5747346/)

#load('/home/rstudio/dat/session6.Rdata')

# make some output tables for later..
library(dplyr)

counts_and_genes <- inner_join(tibble::rownames_to_column(as.data.frame(ypr$counts), var = 'loci'), tibble::rownames_to_column(as.data.frame(ypr$genes), var = 'loci'))

counts_and_genes$methylation_proportion_1 <- (counts_and_genes$'RH-1-Me') / (counts_and_genes$'RH-1-Me' + counts_and_genes$'RH-1-Un')
counts_and_genes$methylation_proportion_2 <- (counts_and_genes$'RH-2-Me') / (counts_and_genes$'RH-2-Me' + counts_and_genes$'RH-2-Un')
counts_and_genes$methylation_proportion_3 <- (counts_and_genes$'RH-3-Me') / (counts_and_genes$'RH-3-Me' + counts_and_genes$'RH-3-Un')
counts_and_genes$methylation_proportion_4 <- (counts_and_genes$'RH-4-Me') / (counts_and_genes$'RH-4-Me' + counts_and_genes$'RH-4-Un')
counts_and_genes$methylation_proportion_5 <- (counts_and_genes$'RH-5-Me') / (counts_and_genes$'RH-5-Me' + counts_and_genes$'RH-5-Un')
counts_and_genes$methylation_proportion_6 <- (counts_and_genes$'RH-6-Me') / (counts_and_genes$'RH-6-Me' + counts_and_genes$'RH-6-Un')
counts_and_genes$methylation_proportion_7 <- (counts_and_genes$'RH-7-Me') / (counts_and_genes$'RH-7-Me' + counts_and_genes$'RH-7-Un')
counts_and_genes$methylation_proportion_8 <- (counts_and_genes$'RH-8-Me') / (counts_and_genes$'RH-8-Me' + counts_and_genes$'RH-8-Un')
counts_and_genes$methylation_proportion_10 <- (counts_and_genes$'RH-10-Me') / (counts_and_genes$'RH-10-Me' + counts_and_genes$'RH-10-Un')
counts_and_genes$methylation_proportion_11 <- (counts_and_genes$'RH-11-Me') / (counts_and_genes$'RH-11-Me' + counts_and_genes$'RH-11-Un')
counts_and_genes$methylation_proportion_12 <- (counts_and_genes$'RH-12-Me') / (counts_and_genes$'RH-12-Me' + counts_and_genes$'RH-12-Un')
counts_and_genes$methylation_proportion_14 <- (counts_and_genes$'RH-14-Me') / (counts_and_genes$'RH-14-Me' + counts_and_genes$'RH-14-Un')
counts_and_genes$methylation_proportion_15 <- (counts_and_genes$'RH-15-Me') / (counts_and_genes$'RH-15-Me' + counts_and_genes$'RH-15-Un')
counts_and_genes$methylation_proportion_16 <- (counts_and_genes$'RH-16-Me') / (counts_and_genes$'RH-16-Me' + counts_and_genes$'RH-16-Un')
counts_and_genes$methylation_proportion_18 <- (counts_and_genes$'RH-18-Me') / (counts_and_genes$'RH-18-Me' + counts_and_genes$'RH-18-Un')
counts_and_genes$methylation_proportion_19 <- (counts_and_genes$'RH-19-Me') / (counts_and_genes$'RH-19-Me' + counts_and_genes$'RH-19-Un')
counts_and_genes$methylation_proportion_21 <- (counts_and_genes$'RH-21-Me') / (counts_and_genes$'RH-21-Me' + counts_and_genes$'RH-21-Un')
counts_and_genes$methylation_proportion_22 <- (counts_and_genes$'RH-22-Me') / (counts_and_genes$'RH-22-Me' + counts_and_genes$'RH-22-Un')

# Design matrix w a one-way layout

targets$group <- gsub(" ", "", targets$group, fixed = TRUE)
targets$groups <- targets$group 

design <- modelMatrixMeth(model.matrix(~ 0 + groups, data=targets))
design

# Estimate dispersion and model 
y_glm_disp <- estimateDisp(ypr, design=design,  trend="none")
y_glm_fit <- glmFit(y_glm_disp, design)

# Contrasts of interest for our study 

# Research Question 1 - SMV normal vs SMV ischemic 
SMVisch_vs_SMVnormal <- glmLRT(y_glm_fit, contrast = makeContrasts(groupsSMVnormal - groupsSMVischemic, levels=design)) 

# Research Question 2 - HSMV normal vs HSMV ischemic 
HSMVisch_vs_HSMVnormal <- glmLRT(y_glm_fit, contrast = makeContrasts(groupsHSMVnormal - groupsHSMVischemic, levels=design)) 

# Research Question 3 - SMV normal vs SMV ischemic vs HSMV normal vs HSMV ischemic
diffisch_vs_diffnormal <- glmLRT(y_glm_fit, contrast = makeContrasts( (groupsSMVnormal - groupsSMVischemic) - (groupsHSMVnormal - groupsHSMVischemic), levels=design))

# Research Question 4 - MVM vs SMV ischemic 
MVM_vs_SMVischemic <- glmLRT(y_glm_fit, contrast = makeContrasts(groupsMVM - groupsSMVischemic, levels=design)) 

# Research Question 5 - HVM vs HSMV ischemic 
HVM_vs_HSMVischemic <- glmLRT(y_glm_fit, contrast = makeContrasts(groupsHVM - groupsHSMVischemic, levels=design)) 

# Add corrected p-value columns

# Research Question 1 - SMV normal vs SMV ischemic 
SMVisch_vs_SMVnormal$table$BH <- p.adjust(SMVisch_vs_SMVnormal$table$PValue, method = "BH")
SMVisch_vs_SMVnormal$table$bonferroni <- p.adjust(SMVisch_vs_SMVnormal$table$PValue, method = "bonferroni")

# Research Question 2 - HSMV normal vs HSMV ischemic 
HSMVisch_vs_HSMVnormal$table$BH <- p.adjust(HSMVisch_vs_HSMVnormal$table$PValue, method = "BH")
HSMVisch_vs_HSMVnormal$table$bonferroni <- p.adjust(HSMVisch_vs_HSMVnormal$table$PValue, method = "bonferroni")

# Research Question 3 - SMV normal vs SMV ischemic vs HSMV normal vs HSMV ischemic
diffisch_vs_diffnormal$table$BH <- p.adjust(diffisch_vs_diffnormal$table$PValue, method = "BH")
diffisch_vs_diffnormal$table$bonferroni <- p.adjust(diffisch_vs_diffnormal$table$PValue, method = "bonferroni")

# Research Question 4 - MVM vs SMV ischemic 
MVM_vs_SMVischemic$table$BH <- p.adjust(MVM_vs_SMVischemic$table$PValue, method = "BH")
MVM_vs_SMVischemic$table$bonferroni <- p.adjust(MVM_vs_SMVischemic$table$PValue, method = "bonferroni")

# Research Question 5 - HVM vs HSMV ischemic 
HVM_vs_HSMVischemic$table$BH <- p.adjust(HVM_vs_HSMVischemic$table$PValue, method = "BH")
HVM_vs_HSMVischemic$table$bonferroni <- p.adjust(HVM_vs_HSMVischemic$table$PValue, method = "bonferroni")

#################
# Make tables 
#################
#get entrez IDs
entrez_ids <- getBM(attributes = c("ensembl_gene_id", "entrezgene_id"), 
                mart = ss_mart, 
                values = counts_and_genes$ensembl_gene_id,
                filters = 'ensembl_gene_id')



# Question 1 
SMVisch_vs_SMVnormal_results <- data.frame(SMVisch_vs_SMVnormal$table) %>% tibble::rownames_to_column(var = 'ensembl_gene_id')



# add annotations
SMVisch_vs_SMVnormal_results <- left_join(SMVisch_vs_SMVnormal_results, entrez_ids, by = 'ensembl_gene_id')
# add the raw counts
SMVisch_vs_SMVnormal_raw_results <- left_join(SMVisch_vs_SMVnormal_results, counts_and_genes, by = 'ensembl_gene_id')
# add the normalized counts 
SMVisch_vs_SMVnormal_m01_results <- left_join(SMVisch_vs_SMVnormal_results, M01, by = 'ensembl_gene_id')
# filter results based on on nominal p-value
SMVisch_vs_SMVnormal_raw_results_filtered <- SMVisch_vs_SMVnormal_raw_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)
SMVisch_vs_SMVnormal_m01_results_filtered <- SMVisch_vs_SMVnormal_m01_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)

# Question 2 
HSMVisch_vs_HSMVnormal_results <- data.frame(HSMVisch_vs_HSMVnormal$table) %>% tibble::rownames_to_column(var = 'ensembl_gene_id')
# add annotations
HSMVisch_vs_HSMVnormal_results <- left_join(HSMVisch_vs_HSMVnormal_results, entrez_ids, by = 'ensembl_gene_id')
# add the raw counts
HSMVisch_vs_HSMVnormal_raw_results <- left_join(HSMVisch_vs_HSMVnormal_results, counts_and_genes, by = 'ensembl_gene_id')
# add the normalized counts 
HSMVisch_vs_HSMVnormal_m01_results <- left_join(HSMVisch_vs_HSMVnormal_results, M01, by = 'ensembl_gene_id')
# filter results based on on nominal p-value
HSMVisch_vs_HSMVnormal_raw_results_filtered <- HSMVisch_vs_HSMVnormal_raw_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)
HSMVisch_vs_HSMVnormal_m01_results_filtered <- HSMVisch_vs_HSMVnormal_m01_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)

# Question 3 
diffisch_vs_diffnormal_results <- data.frame(diffisch_vs_diffnormal$table) %>% tibble::rownames_to_column(var = 'ensembl_gene_id')
# add annotations
diffisch_vs_diffnormal_results <- left_join(diffisch_vs_diffnormal_results, entrez_ids, by = 'ensembl_gene_id')
# add the raw counts
diffisch_vs_diffnormal_raw_results <- left_join(diffisch_vs_diffnormal_results, counts_and_genes, by = 'ensembl_gene_id')
# add the normalized counts 
diffisch_vs_diffnormal_m01_results <- left_join(diffisch_vs_diffnormal_results, M01, by = 'ensembl_gene_id')
# filter results based on on nominal p-value
diffisch_vs_diffnormal_raw_results_filtered <- diffisch_vs_diffnormal_raw_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)
diffisch_vs_diffnormal_m01_results_filtered <- diffisch_vs_diffnormal_m01_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)

# Question 4 
MVM_vs_SMVischemic_results <- data.frame(MVM_vs_SMVischemic$table) %>% tibble::rownames_to_column(var = 'ensembl_gene_id')
# add annotations
MVM_vs_SMVischemic_results <- left_join(MVM_vs_SMVischemic_results, entrez_ids, by = 'ensembl_gene_id')
# add the raw counts
MVM_vs_SMVischemic_raw_results <- left_join(MVM_vs_SMVischemic_results, counts_and_genes, by = 'ensembl_gene_id')
# add the normalized counts 
MVM_vs_SMVischemic_m01_results <- left_join(MVM_vs_SMVischemic_results, M01, by = 'ensembl_gene_id')
# filter results based on on nominal p-value
MVM_vs_SMVischemic_raw_results_filtered <- MVM_vs_SMVischemic_raw_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)
MVM_vs_SMVischemic_m01_results_filtered <- MVM_vs_SMVischemic_m01_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)

# Question 5 
HVM_vs_HSMVischemic_results <- data.frame(HVM_vs_HSMVischemic$table) %>% tibble::rownames_to_column(var = 'ensembl_gene_id')
# add annotations
HVM_vs_HSMVischemic_results <- left_join(HVM_vs_HSMVischemic_results, entrez_ids, by = 'ensembl_gene_id')
# add the raw counts
HVM_vs_HSMVischemic_raw_results <- left_join(HVM_vs_HSMVischemic_results, counts_and_genes, by = 'ensembl_gene_id')
# add the normalized counts 
HVM_vs_HSMVischemic_m01_results <- left_join(HVM_vs_HSMVischemic_results, M01, by = 'ensembl_gene_id')
# filter results based on on nominal p-value
HVM_vs_HSMVischemic_raw_results_filtered <- HVM_vs_HSMVischemic_raw_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)
HVM_vs_HSMVischemic_m01_results_filtered <- HVM_vs_HSMVischemic_m01_results  %>% dplyr::filter(PValue < 0.01) %>% arrange(logFC)

save.image('/home/rstudio/dat/session7.Rdata')

```



```{r}
# <!-- #here's an example of how to make heatmaps from the normalized data tables: -->
# 
# <!-- library(ggplot2) -->
# <!-- library(viridis) -->
# 
# <!-- SMVisch_vs_SMVnormal_norm_results_filtered_long <- pivot_longer(as.data.frame(SMVisch_vs_SMVnormal_norm_results_filtered), cols = !(c(ensembl_gene_id,logFC,logCPM,LR,PValue,BH,bonferroni,entrezgene_id,entrezgene_accession,vgnc)), names_to = 'normalized_methylation') -->
# 
# <!-- labels <- targets -->
# 
# <!-- labels$normalized_methylation <- c("1-Me","2-Me","3-Me","4-Me","5-Me","6-Me","7-Me","8-Me","10-Me","11-Me","12-Me","14-Me","15-Me","16-Me","18-Me","19-Me","21-Me","22-Me") -->
# 
# <!-- SMVisch_vs_SMVnormal_norm_results_filtered_long <- inner_join(SMVisch_vs_SMVnormal_norm_results_filtered_long, labels, by = 'normalized_methylation') -->
# 
# <!-- SMVisch_vs_SMVnormal_norm_results_filtered_long$normalized_methylation <- factor(SMVisch_vs_SMVnormal_norm_results_filtered_long$normalized_methylation, levels = c("1-Me","2-Me","3-Me","4-Me","5-Me","6-Me","7-Me","8-Me","10-Me","11-Me","12-Me","14-Me","15-Me","16-Me","18-Me","19-Me","21-Me","22-Me")) -->
# 
# <!-- ggplot(data = (SMVisch_vs_SMVnormal_norm_results_filtered_long %>% filter(group == "SMVischemic" | group == "SMVnormal") %>% arrange(value)), -->
# <!--        mapping = aes( -->
# <!--          x = normalized_methylation,  -->
# <!--          y = entrezgene_accession,  -->
# <!--          fill = value)) + -->
# <!--   geom_tile() +  -->
# <!--   #scale_fill_distiller(palette = "RdYlBu") + -->
# <!--   scale_fill_viridis(option="viridis") + -->
# <!--   facet_grid(~group, scales = 'free') -->

```

<!-- # Write worksheets/excel -->

<!-- #First, make the readme tab: -->
<!-- readme_glm <- "glmFit, glmLRT, and makeContrasts functions in edgeR were used to fit a genewise negative binomial GLM and make comparisons of interest based on the research question." -->
<!-- readme_loci <- "`loci` column is the location of a given genomic location. The first number or letter is the chromosome, followed by a `-` symbol and a second number indicates the position or location on the chromosome." -->
<!-- readme_logFC <- "`logFC` indicates the log2-fold change of expression between the two conditions tested. For `SMVisch_vs_SMVnormal_results_filtered` results, positive values indicate higher methylation rates in SMVnormal relative to SMVisch, negative values indicate lower methylation rates in SMVnormal vs SMVisch. For `HSMVisch_vs_HSMVnormal_results_filtered` results, positive values indicate higher methylation rates in HSMVnormal relative to HSMVisch, negative values indicate lower methylation rates in HSMVnormal relative to HSMVisch. For `diffisch_vs_diffnormal_results_filtered` results, positive values indicate ...? For `MVM_vs_SMVischemic_results_filtered` results, positive values indicate higher methylation rates in MVM relative to SMVischemic, negative values indicate lower methylation rates in MVM relative to SMVischemic." -->
<!-- readme_logCPM <- "`logCPM` is the average log2-counts per million, the average taken over all libraries used in the experiment." -->
<!-- readme_LR <- "`LR` is the likelihood ratio statistics (larger LR, smaller p-value" -->
<!-- readme_PValue <- "`PValue` probability of obtaining results at least as extreme as the results actually observed, under the assumption that the null hypothesis is correct" -->
<!-- readme_BH <- "`BH` is Benjamini & Hochberg corrected p-values" -->
<!-- readme_bonferroni <- "`bonferroni` Bonferroni corrected p-values" -->
<!-- readme_MeUn <- "The columns with the format `#-Me` or `#-Un` indicate the methylated and unmethylated counts at a given locus for a particular sample." -->
<!-- readme_Chr <- "`Chr` is the chromosome of the locus" -->
<!-- readme_Locus <- "`Locus` is the position of the locus on the chromosome indicated in the `Chr` column." -->
<!-- readme_methylation_proportion <- "The columns with the format `methylated_proportion_#` indicate the proportion of methylated counts at a given loci for a particular sample, calculated as `methylated counts / (unmethylated counts + methylated counts)` for each sample and loci." -->
<!-- readme_TSS_start <- "`TSS_start` indicates the TSS location nearest to the methylation loci shown." -->
<!-- readme_uniprot_gn_symbol <- "`uniprot_gn_symbol` is the gene symbol (if available) associated with each TSS" -->
<!-- readme_ensembl_gene_id <- "`ensembl_gene_id` is the Ensembl gene ID associated with each TSS" -->
<!-- readme_ensembl_gene_start <- "`gene_start` and `gene_end` are the start and end genomic positions for each gene." -->

<!-- readme_sheet <- rbind(readme_glm, readme_loci, readme_logFC, readme_logCPM, readme_LR, readme_PValue, readme_BH, readme_bonferroni, readme_MeUn, readme_Chr, readme_Locus, readme_methylation_proportion, readme_TSS_start, readme_uniprot_gn_symbol, readme_ensembl_gene_id, readme_ensembl_gene_start) -->

<!-- list_1 <- list(readme_sheet, SMVisch_vs_SMVnormal_results_filtered) -->
<!-- names(list_1) <- c("readme", "SMVisch_vs_SMVnormal") -->
<!-- write.xlsx(list_1, file = '/home/rstudio/dat/SMVisch_vs_SMVnormal_results_filtered.xlsx') -->

<!-- list_2 <- list(readme_sheet, HSMVisch_vs_HSMVnormal_results_filtered) -->
<!-- names(list_2) <- c("readme", "HSMVisch_vs_HSMVnormal") -->
<!-- write.xlsx(list_2, file = '/home/rstudio/dat/HSMVisch_vs_HSMVnormal_results_filtered.xlsx') -->

<!-- list_3 <- list(readme_sheet, diffisch_vs_diffnormal_results_filtered) -->
<!-- names(list_3) <- c("readme", "diffisch_vs_diffnormal") -->
<!-- write.xlsx(list_3, file = '/home/rstudio/dat/diffisch_vs_diffnormal_results_filtered.xlsx') -->

<!-- list_4 <- list(readme_sheet, HVM_vs_HSMVischemic_results_filtered) -->
<!-- names(list_4) <- c("readme", "HVM_vs_HSMVischemic") -->
<!-- write.xlsx(list_4, file = '/home/rstudio/dat/HVM_vs_HSMVischemic_results_filtered.xlsx') -->




